### PR TITLE
Validate account balance creation/update using DTO

### LINKED ISSUE #3341: Adopt validation with DTO for account balance management
Adopt concept of form validation with DTO for the account balance management.

Related issue: https://github.com/ghostfolio/ghostfolio/issues/3294

### PR DESCRIPTION
**Pull request by Fedron**
Closes #3341

Update data service to use the `CreateAccountBalanceDto`.

Refactor account balance component to validate the form using the DTO. The validated DTO is then emitted.

Change account detail dialog to use`CreateAccountBalanceDto`.

### CODE REVIEW

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts [hunk 1]
```diff
@@ -95,19 +96,9 @@ export class AccountDetailDialog implements OnDestroy, OnInit {
...
   }
 
-  public onAddAccountBalance({
-    balance,
-    date
-  }: {
-    balance: number;
-    date: Date;
-  }) {
+  public onAddAccountBalance(balance: CreateAccountBalanceDto) {
```
Comment (dtslvr) [inline apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts:99]: Can you refactor to `accountBalance: CreateAccountBalanceDto`?

File: apps/client/src/app/services/data.service.ts [hunk 2]
```diff
@@ -618,20 +619,8 @@ export class DataService {
...
-    accountId: string;
-    balance: number;
-    date: Date;
-  }) {
-    return this.http.post<AccountBalance>(`/api/v1/account-balance`, {
-      accountId,
-      balance,
-      date
-    });
+  public postAccountBalance(aBalance: CreateAccountBalanceDto) {
```
Comment (dtslvr) [inline apps/client/src/app/services/data.service.ts:622]: `aAccountBalance: CreateAccountBalanceDto`

File: libs/ui/src/lib/account-balances/account-balances.component.ts [hunk 3]
```diff
@@ -107,8 +106,25 @@ export class GfAccountBalancesComponent
     }
   }
 
-  public onSubmitAccountBalance() {
-    this.accountBalanceCreated.emit(this.accountBalanceForm.getRawValue());
+  public async onSubmitAccountBalance() {
+    const balance: CreateAccountBalanceDto = {
```
Comment (dtslvr) [inline libs/ui/src/lib/account-balances/account-balances.component.ts:110]: `const accountBalance`...

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks @Fedron for the contribution. I have added a minor suggestion for refactoring.

File: apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts (hunk 1 above)
Comment (Fedron) [reply apps/client/src/app/components/account-detail-dialog/account-detail-dialog.component.ts]: Changed

File: apps/client/src/app/services/data.service.ts (hunk 2 above)
Comment (Fedron) [reply apps/client/src/app/services/data.service.ts]: Changed

File: libs/ui/src/lib/account-balances/account-balances.component.ts (hunk 3 above)
Comment (Fedron) [reply libs/ui/src/lib/account-balances/account-balances.component.ts]: Changed

**Review by dtslvr — APPROVED** [review]
Thanks again for your contribution, @Fedron 😎

**Pull request merged**